ci/build_docs.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+set -euo pipefail
+
+rapids-logger "Create test conda environment"
+. /opt/conda/etc/profile.d/conda.sh
+
+rapids-dependency-file-generator \
+  --output conda \
+  --file_key docs \
+  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION}" | tee env.yaml
+
+rapids-mamba-retry env create --force -f env.yaml -n test
+conda activate test
+
+rapids-print-env
+
+rapids-logger "Downloading artifacts from previous jobs"
+CPP_CHANNEL=$(rapids-download-conda-from-s3 cpp)
+PYTHON_CHANNEL=$(rapids-download-conda-from-s3 python)
+VERSION_NUMBER=$(rapids-get-rapids-version-from-git)
+
+rapids-mamba-retry install \
+  --channel "${CPP_CHANNEL}" \
+  --channel "${PYTHON_CHANNEL}" \
+  libcuspatial \
+  cuspatial
+
+
+# Build Doxygen docs
+gpuci_logger "Build Doxygen docs"
+pushd cpp/doxygen
+doxygen Doxyfile
+popd
+
+# Build Python docs
+gpuci_logger "Build Sphinx docs"
+pushd docs
+sphinx-build -b dirhtml source _html
+sphinx-build -b text source _text
```

Comment:
```suggestion
sphinx-build -b dirhtml source _html -W
sphinx-build -b text source _text -W
```

@AjayThorve added this flag in https://github.com/rapidsai/cuml/pull/5155/files, which I think is a good idea to add to all of the repositories.

It will treat warnings as errors as indicated by the sphinx docs: https://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-W

If any errors appear as a result of this, we can try to resolve them ourselves or solicit feedback from the cuspatial team.